File: packages/go-sdk/operations/health.go
package operations

import "context"

// HealthService provides health check operations.
type HealthService struct {
	r Requester
}

// NewHealthService creates a new HealthService.
func NewHealthService(r Requester) *HealthService {
	return &HealthService{r: r}
}

// List performs a liveness check.
func (s *HealthService) List(ctx context.Context) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "GET", "/health", nil, nil, nil, &result)
	return result, err
}

// GetReady performs a readiness check.
func (s *HealthService) GetReady(ctx context.Context) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "GET", "/health/ready", nil, nil, nil, &result)
	return result, err
}

// ListMetrics returns Prometheus metrics.
func (s *HealthService) ListMetrics(ctx context.Context) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "GET", "/metrics", nil, nil, nil, &result)
	return result, err
}

File: packages/rust-sdk/src/operations/health.rs
use crate::client::StraitClient;
use crate::errors::StraitError;
use std::sync::Arc;

pub struct HealthService {
    client: Arc<StraitClient>,
}

impl HealthService {
    pub fn new(client: Arc<StraitClient>) -> Self {
        Self { client }
    }

    pub async fn list(&self) -> Result<serde_json::Value, StraitError> {
        self.client.do_request("GET", "/v1/health", None, None, None).await
    }

    pub async fn get_ready(&self) -> Result<serde_json::Value, StraitError> {
        self.client.do_request("GET", "/v1/health/ready", None, None, None).await
    }

    pub async fn list_metrics(&self) -> Result<serde_json::Value, StraitError> {
        self.client.do_request("GET", "/v1/health/metrics", None, None, None).await
    }
}
